socorro/external/postgresql/adi.py

```diff
@@ -0,0 +1,84 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+import logging
+
+from socorro.external.postgresql.base import PostgreSQLBase
+from socorro.external import MissingArgumentError
+from socorro.lib import external_common
+
+logger = logging.getLogger("webapi")
+
+
+class ADI(PostgreSQLBase):
+
+    def get(self, **kwargs):
+        yesterday = datetime.datetime.utcnow() - datetime.timedelta(1)
+        tomorrow = yesterday + datetime.timedelta(2)
+        yesterday = yesterday.date()
+        tomorrow = tomorrow.date()
+        filters = [
+            ('start_date', yesterday, 'date'),
+            ('end_date', tomorrow, 'date'),
+            ('product', '', 'str'),
+            ('version', '', 'str'),
+        ]
+        params = external_common.parse_arguments(filters, kwargs)
+        required = (
+            'start_date',
+            'end_date',
+            'product',
+            'version',
+        )
+        missing = []
+        for each in required:
+            if not params.get(each):
+                missing.append(each)
+        if missing:
+            raise MissingArgumentError(', '.join(missing))
+
+        SQL = """
+        SELECT
+            SUM(adi_count) AS adi_count,
+            date::DATE,
+            product_name,
+            product_os_platform,
+            product_version,
+            update_channel
+        FROM
+            raw_adi
+        WHERE
+            product_name = %(product)s AND
+            product_version = %(version)s AND
+            date BETWEEN %(start_date)s::DATE AND %(end_date)s::DATE
+        GROUP BY
+            date::DATE,
+            product_name,
+            product_os_platform,
+            product_version,
+            update_channel
+        """
+
+        results = self.query(SQL, params)
+
+        fields = (
+            'adi_count',
+            'date',
+            'product_name',
+            'product_os_platform',
```

Comment:
So if I understood our "path" correctly, in the near future we are going to use this service along with Super Search to build our graphs. What worries me here is that this field seems to contain data that is not available from Super Search: the "platform" can be one of "linux", "mac os x" or "windows nt" and rarely anything else. I don't think we have the more precise OS names like "Darwin" or "SunOS", so I'm concerned that we might be confused when trying to filter by platform. If it's inconsistent we should either make it consistent (either by putting the appropriate data in ES, through the processors like we did with the beta magic numbers or by making it simpler in our Postgres tables) or we do not expose this at all.